recipes/select-first-n-elements-from-list.md

```diff
@@ -0,0 +1,26 @@
+# Select first n elements from list
+
+## Problem
+
+I want to select first elements from list.
+
+## Solution
+
+```schem
+(define (take lst n)
+  (let loop ((result '()) (i n) (lst lst))
+    (if (or (null? lst) (<= i 0))
```

Comment:
[SRFI 1 `take`](https://srfi.schemers.org/srfi-1/srfi-1.html#take) doesn't say what happens when the list has less than `n` elements. [The sample implementation](https://github.com/scheme-requests-for-implementation/srfi-1/blob/41993e344c1f5557cc125e9696cd746d2a1e46ac/srfi-1-reference.scm#L482) implicitly raises an exception since `cdr` will fail on a non-list. I also tried `(import (srfi 1))` in three Scheme implementations, and they all raise an exception. Hence, our `take` should probably do so too.

Reply:
I can make two variations if you want. Because I think that is more useful:

```scheme
(take '(1 2) 10)
;; ==> (1 2)
```